### PR TITLE
Changes after rebase an already approved PR:24767

### PR DESCRIPTION
**Pull request by rsubra13**
##### SUMMARY
Rebase changes after an already approved PR:24767

##### ISSUE TYPE
 - Feature Pull Request
 
##### COMPONENT NAME
ec2_ami_copy

##### ANSIBLE VERSION
ansible 2.5.2

##### ADDITIONAL INFORMATION
Nil

### CODE REVIEW

File: lib/ansible/modules/cloud/amazon/ec2_ami_copy.py [hunk 1]
```diff
@@ -39,17 +39,24 @@
...
     description:
       - Whether or not the destination snapshots of the copied AMI should be encrypted.
+    required: false
+    default: null
     version_added: "2.2"
   kms_key_id:
     description:
       - KMS key id used to encrypt image. If not specified, uses default EBS Customer Master Key (CMK) for your account.
+    required: false
+    default: null
```
Comment (ryansb) [inline lib/ansible/modules/cloud/amazon/ec2_ami_copy.py]: No need to re-add these docs, as the default for any non-specified param is `null`

File: lib/ansible/modules/cloud/amazon/ec2_ami_copy.py [hunk 2]
```diff
@@ -160,27 +175,33 @@ def copy_image(module, ec2):
...
+            image = ec2.copy_image(**params)
+            image_id = image['ImageId']
+            if tags:
+                ec2.create_tags(Resources=[image_id],
+                                Tags=ansible_dict_to_boto3_tag_list(tags))
+            changed = True
+
         if module.params.get('wait'):
-            delay = 15
+            ec2.get_waiter('image_available').wait(ImageIds=[image['ImageId']])
```
Comment (ryansb) [inline lib/ansible/modules/cloud/amazon/ec2_ami_copy.py]: No need to wait twice - the below lines do the same "wait" while respecting max wait time.

**Review by ryansb — CHANGES_REQUESTED** [review]

**Review by s-hertel — APPROVED** [review]

File: lib/ansible/modules/cloud/amazon/ec2_ami_copy.py [hunk 3]
```diff
@@ -39,13 +39,16 @@
...
       - The name of the new AMI to copy. (As of 2.3 the default is 'default', in prior versions it was 'null'.)
+    required: false
     default: "default"
   description:
     description:
       - An optional human-readable string describing the contents and purpose of the new AMI.
   encrypted:
     description:
       - Whether or not the destination snapshots of the copied AMI should be encrypted.
+    required: false
```
Comment (ryansb) [inline lib/ansible/modules/cloud/amazon/ec2_ami_copy.py]: These lines should also be removed

File: lib/ansible/modules/cloud/amazon/ec2_ami_copy.py [hunk 4]
```diff
@@ -196,10 +210,11 @@ def main():
...
-        tags=dict(type='dict')))
-
-    module = AnsibleModule(argument_spec=argument_spec)
+        tags=dict(type='dict')),
+        tag_equality=dict(type='bool', default=False))
 
+    module = AnsibleAWSModule(argument_spec=argument_spec)
+    # TODO: Check botocore version
     region, ec2_url, aws_connect_params = get_aws_connection_info(module, boto3=True)
     ec2 = boto3_conn(module, conn_type='client', resource='ec2', region=region, endpoint=ec2_url,
```
Comment (ryansb) [inline lib/ansible/modules/cloud/amazon/ec2_ami_copy.py]: Also, these 2 lines can be replaced with `ec2 = module.client('ec2')` and the boto3_conn and get_aws_connection_info imports can be removed.

**Review by ryansb — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/cloud/amazon/ec2_ami_copy.py (hunk 3 above)
Comment (rsubra13) [reply lib/ansible/modules/cloud/amazon/ec2_ami_copy.py]: Removed them! Thanks @ryansb

File: lib/ansible/modules/cloud/amazon/ec2_ami_copy.py (hunk 4 above)
Comment (rsubra13) [reply lib/ansible/modules/cloud/amazon/ec2_ami_copy.py]: Done! Thanks.

**Review by rsubra13 — COMMENTED** [review]
Finished with the changes!

**Review by ryansb — APPROVED** [review]

**Comment by s-hertel** [comment]
Thanks for your first contribution @rsubra13!

**Pull request merged**